src/detect-mqtt-connack-sessionpresent.c

```diff
@@ -0,0 +1,328 @@
+/* Copyright (C) 2020 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+/**
+ * \file
+ *
+ * \author Sascha Steinbiss <sascha@steinbiss.name>
+ */
+
+#include "suricata-common.h"
+#include "conf.h"
+#include "detect.h"
+#include "detect-parse.h"
+#include "detect-engine.h"
+#include "detect-engine-content-inspection.h"
+#include "detect-mqtt-connack-sessionpresent.h"
+#include "util-unittest.h"
+
+#include "rust-bindings.h"
+
+#define PARSE_REGEX "^true|false|yes|no$"
+static DetectParseRegex parse_regex;
+
+static int mqtt_connack_session_present_id = 0;
+
+static int DetectMQTTConnackSessionPresentMatch(DetectEngineThreadCtx *det_ctx,
+                               Flow *f, uint8_t flags, void *state,
+                               void *txv, const Signature *s,
+                               const SigMatchCtx *ctx);
+static int DetectMQTTConnackSessionPresentSetup (DetectEngineCtx *, Signature *, const char *);
+void MQTTConnackSessionPresentRegisterTests(void);
+void DetectMQTTConnackSessionPresentFree(void *);
+
+static int DetectEngineInspectMQTTConnackSessionPresentGeneric(ThreadVars *tv,
+        DetectEngineCtx *de_ctx, DetectEngineThreadCtx *det_ctx,
+        const Signature *s, const SigMatchData *smd,
+        Flow *f, uint8_t flags, void *alstate,
+        void *txv, uint64_t tx_id);
+
+typedef struct DetectMQTTConnackSessionPresentData_ {
```

Comment:
Why wrap a single bool into a `struct` ?

Reply:
It made handling the boilerplate code easier for that many keywords. But I can polish it a bit and make it more idiomatic.

Reply:
Done.